lms/djangoapps/branding_by_referer/branding_by_referer.py

```diff
@@ -79,8 +79,17 @@ def get_stored_referer_data(self, request):
             referer_on_cookie = request.COOKIES.get(self.COOKIE_MARKETING_SITE_REFERER, None)
 
             if stored_referer_data and referer_on_cookie:
-                stored_referer_data_json = json.loads(stored_referer_data)
-                stored_referer_data_json['referer_domain'] = referer_on_cookie
+                try:
+                    stored_referer_data_json = json.loads(stored_referer_data)
+                    stored_referer_data_json['referer_domain'] = referer_on_cookie
+                except ValueError:
+                    # This is for support legacy user preferences records,
+                    # that have a string value instead of json value.
+                    stored_referer_data_json = {
+                        'referer_domain': referer_on_cookie,
+                        'site_domain': request.get_host()
```

Comment:
I think we could use 'django.contrib.sites.shortcuts.get_current_site()' instead

Reply:
I would say leave as it is, just beacuse get_current_site uses request.get_host() and to prevent changing the code in other places